Review firewall requirements (#4210)
theforeman/foreman-documentation: guides/common/modules/con_smart-proxy-networking.adoc

Reviewer: For this chapter we run into the naming problem: Capsule in downstream is IMHO something quite different than a Smart Proxy.

My interpretation:
* Smart Proxy: the generic concept that Foreman connects to
* Foreman Proxy: the Ruby implementation of Smart Proxy
* Capsule: The downstream naming for a server that has a few services: Foreman Proxy, Pulp, Apache (for the RHSM registration API). Upstream we sometimes refer to this as Content Proxies: a Proxy with a set of content features.

Now looking ahead we're going to make this a bigger problem. The Insights on Premise is also recognized as a Smart Proxy. Then with [pulp_smart_proxy](https://github.com/theforeman/pulp_smart_proxy) the Pulp server can identify as a Smart Proxy.

In other words: I think soon we need to have the discussion within Red Hat: where do we want to go with this. Do we stop branding Smart Proxy as Capsule?

[id="{smart-proxy-context}-networking"]
= {SmartProxy} networking

The communication between {ProjectServer} and hosts registered to a {SmartProxyServer} is routed through that {SmartProxyServer}.
{SmartProxyServer} also provides {Project} services to hosts.

[NOTE]
====
Some outgoing traffic returns to {Project} to enable internal communication and security operations.
====

> 

Author: I'd like to resolve this here so that we're sure the reviewed requirements are okay. @evgeni can you shed some light on this? Can we perhaps specify the note? Or is it safe to drop it?

> 

Reviewer 2: Well, it was added in https://github.com/theforeman/foreman-documentation/pull/2451 and is frankly a tad confusing.
So strap on your best glasses, we're using a Tardis!

The bug that #2451 was trying to fix (https://bugzilla.redhat.com/show_bug.cgi?id=2233266) was reported against Satellite 6.13, and we look at https://docs.redhat.com/en/documentation/red_hat_satellite/6.13/html/installing_satellite_server_in_a_connected_network_environment/preparing_your_environment_for_installation_satellite#Ports_and_Firewalls_Requirements_satellite we see the following entries with "Satellite" as the destination:

| Destination Port | Protocol | Service | Destination | Required For | Description |
|-----------------------|-------------|-----------|----------------|------------------|----------------|
| 443 | TCP | HTTPS | Satellite | Capsule | Capsule, Configuration management, Template retrieval, OpenSCAP, Remote Execution result upload |
|  5646 | TCP | AMQP | Satellite Server | Katello agent | Forward message to Qpid dispatch router on Capsule (optional) |
| 5671 | | | Satellite Server | Remote install for Katello agent | Send install command to client |
| 5671 | | | Satellite Server | Remote install for Katello agent | Forward message to dispatch router on Satellite |

(Yes, protocol and service are missing for the last two lines, yes "Satellite" vs "Satellite Server" is inconsistent, have a :wine_glass: for me please)

Now, it is true that the way Katello agent communication was set up, the Germans would call "from behind, through the chest, into the eye" (Sorry Anet, if you don't have a visual picture of the pain, I either owe you one, or you can count yourself lucky, or both).

Luckily for us, today, stepped out of the Tardis again: Katello agent is a sin of the past and we don't have to care about it anymore and can solely focus on the one remaining entry in todays docs:

| Destination Port | Protocol | Service | Destination | Required For | Description |
|-----------------------|-------------|-----------|----------------|------------------|----------------|
| 443 | TCP | HTTPS | Satellite | Capsule | Capsule, Configuration management, Template retrieval, OpenSCAP, Remote Execution result upload |

And the great thing? It's nonsense! I mean, yeah, the external Capsule needs to be able to talk to the Satellite on port 443 (and others!) but that should be covered by "{Project} **incoming** traffic" and "**{SmartProxy}** outgoing traffic" and never be part of "{Project} outgoing traffic"

So, my verdict is:
* drop that note
* drop line 65-73 of `guides/common/modules/ref_project-server-port-and-firewall-requirements.adoc` (`| 443 | TCP | HTTPS | {Project} | {SmartProxy} | {SmartProxy} … `)

ifndef::satellite[]
In a topology with hosts connecting to a {SmartProxyServer}, 
endif::[]
ifdef::satellite[]
In xref:{project-context}-topology-with-hosts-connecting-to-a-{smart-proxy-context}[], 
endif::[]

> {SmartProxy} provides a single endpoint for all host network communications so that in remote network segments, only firewall ports to the {SmartProxy} itself must be open.

Author: I did s/SmartProxy/SmartProxyServer/ but as for the other part of your suggestion, either I don't understand what you mean or I don't think it's an improvement. To me it makes more sense as it is than what you suggests. Please let me know if I'm missing something.

> Hosts do not need direct access to the {ProjectServer} itself.

Reviewer 3: ```suggestion
Hosts do not need direct access to {ProjectServer}.
```

// TODO: Replace graphic with simpler graphic and reference to "Port and firewall requirements"
ifdef::satellite[]
[id="{project-context}-topology-with-hosts-connecting-to-a-{smart-proxy-context}"]
.{Project} topology with hosts connecting to a {SmartProxy}
image::common/topology-isolated-satellite.png[{ProjectName} topology with a host]
endif::[]

ifndef::satellite[]
In a topology with hosts connecting directly to {ProjectServer}, 
endif::[]
ifdef::satellite[]
In xref:{project-context}-topology-with-hosts-connecting-directly-to-{project-context}-server[], 
endif::[]

> hosts connect to {ProjectServer} rather than a {SmartProxy}.

Reviewer 3: IMHO it's both technically wrong (because hosts connect to the "integrated" Smart Proxy of Foreman Server), and not providing any insight ("If hosts do X, they do X not Y").

Author: I believe the point here is compare the two setups. From that perspective, to me it makes sense to say that in the first one "hosts connect to proxy" and in the second one "host don't connect to proxy". Can you see an alternative?

Author: Okay, I tried rewording it. Let me know what you think.

Reviewer 3: Great. Thx.

> This applies also to {SmartProxies} themselves because the {SmartProxyServer} is a host of {ProjectServer}.

Reviewer 3: ```suggestion
This applies to all {SmartProxyServers} because they are hosts of {ProjectServer}.
```

> Hosts need direct network access to {ProjectServer}.

Reviewer 3: In general, hosts need access to _their_ Smart Proxy; which can be either Foreman Server or a Smart Proxy Server.

Author: Yes, but this is a description of networking requirements for the setup where you don't deploy a proxy. Which means that the sentence is valid.

[id="{smart-proxy-context}-networking"]
= {SmartProxy} networking

The communication between {ProjectServer} and hosts registered to a {SmartProxyServer} is routed through that {SmartProxyServer}.
{SmartProxyServer} also provides {Project} services to hosts.

> Integrated {SmartProxy}::

Reviewer: This is a concept I'd love to get rid of. It's confusing people and serves no purpose IMHO.

We can wait until we containerize, but as it's looking now we will not provide a Foreman Proxy on the `{ProjectServer}`. At least, not by default.

Author: I think the section will still work without the definitions of integrated/external smart proxies. So that's what I did -- I dropped them.

Integrated {SmartProxy} is the {SmartProxy} contained in a {ProjectServer}.

> A host that is directly connected to a {ProjectServer} is considered a client of its integrated {SmartProxy}.

Reviewer: IMHO a host isn't a client of "its integrated {SmartProxy}". Especially in the non-Katello context that's technically false: hosts don't connect to `{SmartProxy}`.
